src/writeData/subscriptions/context/subscription.create.tsx

```diff
@@ -0,0 +1,149 @@
+// Libraries
+import React, {FC, useState, useCallback, useEffect} from 'react'
+import {createAPI} from 'src/writeData/subscriptions/context/api'
+import {useSelector} from 'react-redux'
+import {useHistory} from 'react-router-dom'
+
+// Utils
+import {SUBSCRIPTIONS, LOAD_DATA} from 'src/shared/constants/routes'
+import {getOrg} from 'src/organizations/selectors'
+import {sanitizeForm} from '../utils/form'
+
+// Types
+import {RemoteDataState} from 'src/types'
+import {Subscription} from 'src/types/subscriptions'
+
+export interface SubscriptionCreateContextType {
+  create: () => void
+  formContent: Subscription
+  formComplete: boolean
+  setFormComplete: (boolean) => void
+  updateForm: (formContent) => void
+  loading: RemoteDataState
+}
+
+export const DEFAULT_CONTEXT: SubscriptionCreateContextType = {
+  create: () => {},
+  formComplete: false,
+  formContent: {
+    name: '',
+    description: '',
+    protocol: 'MQTT',
+    brokerHost: '',
+    brokerPort: 0,
+    brokerUsername: '',
+    brokerPassword: '',
+    brokerCert: '',
+    brokerKey: '',
+    topic: '',
+    dataFormat: 'lineprotocol',
+    jsonMeasurementKey: {
+      name: '',
+      path: '',
+      type: 'string',
+    },
+    jsonFieldKeys: [
+      {
+        name: '',
+        path: '',
+        type: 'string',
+      },
+    ],
+    jsonTagKeys: [
+      {
+        name: '',
+        path: '',
+        type: 'string',
+      },
+    ],
+    jsonTimestamp: {
+      name: '',
+      path: '',
+      type: 'string',
+    },
+    stringMeasurement: {
+      pattern: '',
+      name: '',
+    },
+    stringFields: [
+      {
+        pattern: '',
+        name: '',
+      },
+    ],
+    stringTags: [
+      {
+        pattern: '',
+        name: '',
+      },
+    ],
+    stringTimestamp: {
+      pattern: '',
+      name: '',
+    },
+    bucket: 'nifi',
+    qos: 0,
+  },
+  setFormComplete: () => {},
+  updateForm: () => {},
+  loading: RemoteDataState.NotStarted,
+} as SubscriptionCreateContextType
+
+export const SubscriptionCreateContext = React.createContext<
+  SubscriptionCreateContextType
+>(DEFAULT_CONTEXT)
+
+export const SubscriptionCreateProvider: FC = ({children}) => {
+  const [formContent, setFormContent] = useState(DEFAULT_CONTEXT.formContent)
+  const [formComplete, setFormComplete] = useState(false)
+  const [loading, setLoading] = useState(RemoteDataState.Done)
+  const history = useHistory()
+  const org = useSelector(getOrg)
+  const create = (formContent?: Subscription): any => {
+    setLoading(RemoteDataState.Loading)
+    createAPI({data: formContent})
+      .then(() => {
+        setLoading(RemoteDataState.Done)
+        history.push(`/orgs/${org.id}/${LOAD_DATA}/${SUBSCRIPTIONS}`)
+        console.log('success')
+      })
+      .catch(() => {
```

Comment:
We don't have an error state yet for this - i believe it will be some kind of modal

Reply:
A notification would suffice for now. We do this for notebooks: `dispatch(notify(notebookCreateFail()))` where `notebookCreateFail` is an action that contains a specific failure message

Reply:
ok nice, added!